### PR TITLE
Added demos for text-highlight and wait components

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-moar-demos](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-moar-demos)

File: src/app/docs/text-highlight/text-highlight-docs.component.html [hunk 1]
```diff
@@ -7,17 +7,54 @@
...
-  </sky-docs-demo>
+  <sky-docs-demo>
+    <div
+      class="sky-form-group"
+      style="width:200px;"
+    >
+      <label
+        [for]="searchTermInput.id"
+      >
+        Enter text to highlight:
```
Comment (blackbaud-johnly) [inline src/app/docs/text-highlight/text-highlight-docs.component.html:18]:
```suggestion
        Text to highlight
```
This change is more in line with our guidance at https://developer.blackbaud.com/skyux/design/guidelines/form-design#field-usability.

File: src/app/docs/text-highlight/text-highlight-docs.component.html [hunk 2]
```diff
@@ -7,17 +7,54 @@
...
 
-  <!-- <sky-docs-demo>
-  </sky-docs-demo>
+  <sky-docs-demo>
+    <div
+      class="sky-form-group"
+      style="width:200px;"
+    >
+      <label
+        [for]="searchTermInput.id"
```
Comment (blackbaud-johnly) [inline src/app/docs/text-highlight/text-highlight-docs.component.html:16]:
```suggestion
        [for]="searchTermInput.id"
        class="sky-control-label"
```
The space between the label and input looks too tight, so I think we should apply this class.

File: src/app/public/plugin-resources/code-examples/text-highlight/basic/text-highlight-demo.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<div
+  class="sky-form-group"
+  style="width:200px;"
+>
+  <label
+    [for]="searchTermInput.id"
+  >
+    Enter text to highlight:
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/text-highlight/basic/text-highlight-demo.component.html:8]:
```suggestion
    Text to highlight
```

File: src/app/public/plugin-resources/code-examples/text-highlight/basic/text-highlight-demo.component.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<div
+  class="sky-form-group"
+  style="width:200px;"
+>
+  <label
+    [for]="searchTermInput.id"
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/text-highlight/basic/text-highlight-demo.component.html:6]:
```suggestion
    [for]="searchTermInput.id"
    class="sky-control-label"
```

File: src/app/docs/text-highlight/text-highlight-docs.component.html [hunk 5]
```diff
@@ -7,17 +7,54 @@
...
+    </div>
+
+    <div
+      class="sky-margin-stacked-default"
+    >
+      <sky-checkbox
+        [(ngModel)]="showAdditionalContent"
+      >
+        <sky-checkbox-label>
+          Display additional content
```
Comment (blackbaud-johnly) [inline src/app/docs/text-highlight/text-highlight-docs.component.html:38]: For the demo, it seems like this should be in an *Options* dropdown like the other options we provide to manipulate the demos. Probably not worth worrying about though if this requires a lot of work.

File: src/app/docs/wait/wait-docs.component.html [hunk 6]
```diff
@@ -7,17 +7,43 @@
...
+        >
+        </sky-docs-demo-control-panel-radio-group>
+      </sky-docs-demo-control-panel-section>
+    </sky-docs-demo-control-panel>
+    <button
+      type="button"
+      class="sky-btn sky-btn-default"
+      (click)="onToggleWaitClick()"
+    >
+      Toggle wait for 2 seconds
```
Comment (blackbaud-johnly) [inline src/app/docs/wait/wait-docs.component.html:29]:
```suggestion
      Display wait for 2 seconds
```

File: src/app/public/plugin-resources/code-examples/wait/element/wait-demo.component.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  class="sky-btn sky-btn-default"
+  (click)="isWaiting = !isWaiting"
+>
+  Toggle wait
+</button>
+
+<div
+  style="height: 200px; width: 200px; margin: 10px;"
+>
+  A large area that can be waited with the <code>sky-wait</code> directive.
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/wait/element/wait-demo.component.html:12]:
```suggestion
  The <code>sky-wait</code> directive can apply to a large area.
```

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
LGTM

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-moar-demos](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-moar-demos)

File: src/app/docs/text-highlight/text-highlight-docs.component.html (hunk 5 above)
Comment (Blackbaud-AlexKingman) [reply src/app/docs/text-highlight/text-highlight-docs.component.html]: Good idea. Done.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-moar-demos](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-moar-demos)

**Review by blackbaud-johnly — APPROVED** [review]

**Pull request merged**